ui/apps/platform/src/hooks/useTableSelection.ts

```diff
@@ -55,7 +55,7 @@ function useTableSelection<T extends Base>(
     function getSelectedIds() {
         const ids: string[] = [];
         for (let i = 0; i < selected.length; i += 1) {
-            if (selected[i]) {
+            if (selected[i] && data[i]?.id) {
```

Comment:
This is a good fix for the immediate issue. I looked at the hook again and was considering what causes this issue and seems like its mainly because we do something like this:
```
const { ..., getSelectedIds, ... } = useTableSelection(...);
...
const selectedIds = getSelectedIds();
```
Seems like the `getSelectedIds` function is referencing the wrong `data`. I think it might be better to actually do the `const selectedIds = getSelectedIds();` part within the hook. We would call that function on every render anyways. At least we can prevent this closure issue from happening if we pull it into the hook. The API for that hook would then look something like this:
```
const { ..., getSelectedIds, selectedIds ... } = useTableSelection(...);
...
```
Now the hook will return an updated `selectedIds`, that uses the proper `data` to calculate the selected ids, whenever `data` is updated

Reply:
You don't have to refactor it now, but maybe a `@TODO` for fixing the hook's API in the future would be good to add